### PR TITLE
Fix incorrect dispatch to USAGE in example programs, which causes uninitialized memory to be used

### PR DESCRIPTION
**Pull request by aditya-deshpande-arm**
## Description

Fixes issues #1120 and #6700.

Numerous example programs contain the following:
```
if (argc == 0) {
    goto USAGE;
}

// later in program
p = argv[1];
```
If a program is called from the shell then`argc = 1`, so the program will not go to USAGE due to the check. The access to `argv[1]` is accessing undefined memory. This PR changes the check to `if (argc < 2)`.

## Gatekeeper checklist

- [x] **changelog** provided
- [x] **backport** #7026 
- [x] **tests** not required, example programs

### CODE REVIEW

File: ChangeLog.d/fix-example-programs-no-args.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Bugfix
+   * Fix a bug present in multiple example programs where running the program
+     from the shell without any command line argument results argv[1] being
+     accessed. The above was fixed for the following: pem2der.c, cert_req.c, 
+     cert_app.c, cert_write.c, req_app.c, crl_app.c, dh_genprime.c, key_app.c,
+     gen_key.c, key_app_writer.c, ssl_client2.c, ssl_server2.c, 
+     ssl_mail_client.c.
```
Comment (tom-cosgrove-arm) [inline ChangeLog.d/fix-example-programs-no-args.txt:7]: Probably don't need so much detail, but should mention the issue numbers:

```
   * Fix behavior of sample programs when run with no arguments, which could
     access uninitialized memory in some cases. Fixes #6700 (which was found
     by TrustInSoft Analyzer during REDOCS'22) and #1120.
```

**Comment by tom-cosgrove-arm** [comment]
I suspect the `Lost content` error on `all_u16-check_files` is a race condition while it was assembling the changelog the new version of the file came in, but the assembled changelog didn't include lines from that - i.e. a spurious failure

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Comment by tom-cosgrove-arm** [comment]
CI is failing because of

```
Trailing whitespace:
ChangeLog.d/fix-example-programs-no-args.txt: 3
^^^^check_files: Check: file sanity checks (permissions, encodings): tests/scripts/check_files.py -> 1^^^^
```

**Review by tom-cosgrove-arm — CHANGES_REQUESTED** [review]
Need to fix trailing whitespace

**Comment by tom-cosgrove-arm** [comment]
The Travis failure is due to not having a newline at the end of the ChangeLog file:

```
Missing newline at end of file:
ChangeLog.d/fix-example-programs-no-args.txt
^^^^check_files: Check: file sanity checks (permissions, encodings): tests/scripts/check_files.py -> 1^^^^
```

**Comment by aditya-deshpande-arm** [comment]
> The Travis failure is due to not having a newline at the end of the ChangeLog file:
> 
> ```
> Missing newline at end of file:
> ChangeLog.d/fix-example-programs-no-args.txt
> ^^^^check_files: Check: file sanity checks (permissions, encodings): tests/scripts/check_files.py -> 1^^^^
> ```

Ah, I didn't know that we needed those at the end of non-source files!

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Review by daverodgman — DISMISSED** [review]

**Review by daverodgman — APPROVED** [review]

File: programs/pkey/dh_genprime.c [hunk 2]
```diff
@@ -71,7 +71,7 @@ int main(int argc, char **argv)
     mbedtls_ctr_drbg_init(&ctr_drbg);
     mbedtls_entropy_init(&entropy);
 
-    if (argc == 0) {
+    if (argc < 2) {
```
Comment (gilles-peskine-arm) [inline programs/pkey/dh_genprime.c:74]: No, that's not right. This program doesn't require an argument. Several others don't either. https://github.com/Mbed-TLS/mbedtls/issues/7100

**Pull request merged**